### PR TITLE
Annotation and label can be added with spec/template of RunnerPool

### PR DESCRIPTION
**Pull request by kmdkuk**
#54

### CODE REVIEW

File: api/v1alpha1/runnerpool_types.go [hunk 1]
```diff
@@ -57,6 +57,10 @@ type SlackAgentConfig struct {
 }
 
 type RunnerPodTemplateSec struct {
+	// Standard object's metadata.  Only `annotations` and `labels` are valid.
+	// +optional
+	ObjectMeta `json:"metadata,omitempty"`
```
Comment (ymmt2005) [inline api/v1alpha1/runnerpool_types.go:62]: `,omitempty` here is meaningless.

Comment (kmdkuk) [reply api/v1alpha1/runnerpool_types.go:62]: Thanks for your comment!
Why is that?
I thought that optional fields need `omitempty`, am I wrong?

Comment (ymmt2005) [reply api/v1alpha1/runnerpool_types.go:62]: omitempty is used for JSON serialization, whereas `+optional` declares the filed is optional as OpenAPI schema.
Do you see?

They are different things, and struct (not a pointer) has no `empty` condition, so `omitempty` is meaningless.

Comment (ymmt2005) [reply api/v1alpha1/runnerpool_types.go:62]: For reference, read https://pkg.go.dev/encoding/json#Marshal

> The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string.

Comment (ymmt2005) [reply api/v1alpha1/runnerpool_types.go:62]: If you don't see that, consider that a user can send arbitrary JSON data to kube-apiserver
w/o serializing Go structs.

`+optional` is not related to Go and serialization actually. It is a validation rule for kube-apiserver.

Comment (kmdkuk) [reply api/v1alpha1/runnerpool_types.go:62]: Thank you for the explanation.
I understand the need for `omitempty`.
I will fix it.

File: controllers/runnerpool_controller_test.go [hunk 2]
```diff
@@ -310,6 +317,17 @@ var _ = Describe("RunnerPool reconciler", func() {
...
 		// runner container spec
+		Expect(d.Spec.Template).To(MatchFields(IgnoreExtras, Fields{
+			"ObjectMeta": MatchFields(IgnoreExtras, Fields{
+				"Labels": MatchKeys(IgnoreExtras, Keys{
+					"test-label":                Equal("test"),
+					constants.RunnerOrgLabelKey: Not(Equal("should-not-be-updated")),
+				}),
+				"Annotations": MatchKeys(IgnoreExtras, Keys{
+					"test-annotation": Equal("test"),
+				}),
```
Comment (masa213f) [inline controllers/runnerpool_controller_test.go:328]: Could you check all labels and annotations? I want to check they are merged correctly.

```suggestion
				"Labels": MatchAllKeys(Keys{
					constants.AppNameLabelKey:      Equal(constants.AppName),
					constants.AppComponentLabelKey: Equal(constants.AppComponentRunner),
					constants.AppInstanceLabelKey:  Equal(...),
					constants.RunnerOrgLabelKey:    Equal(...), // 
					constants.RunnerRepoLabelKey:   Equal(...),
					"test-label":                Equal("test"),
				}),
				"Annotations": MatchAllKeys(Keys{
					"test-annotation": Equal("test"),
				}),
```

File: kindtest/manifests/runnerpool1/runnerpool.yaml [hunk 3]
```diff
@@ -8,6 +8,11 @@ spec:
   slackAgent:
     serviceName: slack-agent.meows.svc
   template:
+    metadata:
+      labels:
+        test-label: test
+      annotations:
+        test-annotation: test
```
Comment (masa213f) [inline kindtest/manifests/runnerpool1/runnerpool.yaml:15]: Why did you add those fields?

**Review by zoetrope — APPROVED** [review]
